libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigationProvider.kt

```diff
@@ -0,0 +1,66 @@
+package com.mapbox.navigation.core
+
+import android.content.Context
+import com.mapbox.android.core.location.LocationEngine
+import com.mapbox.android.core.location.LocationEngineProvider
+import com.mapbox.android.core.location.LocationEngineRequest
+import com.mapbox.navigation.base.options.NavigationOptions
+
+/**
+ * Singleton responsible for ensuring there is only one MapboxNavigation instance.
+ */
+object MapboxNavigationProvider {
+    @Volatile
+    private var mapboxNavigation: MapboxNavigation? = null
+
+    /**
+     * Create MapboxNavigation with provided options.
+     * Should be called before [navigation]
+     *
+     * @param context
+     * @param navigationOptions
+     * @param locationEngine
+     * @param locationEngineRequest
+     */
+    @JvmOverloads
+    @JvmStatic
+    fun createNavigation(
+        context: Context,
+        navigationOptions: NavigationOptions,
+        locationEngine: LocationEngine = LocationEngineProvider.getBestLocationEngine(context.applicationContext),
+        locationEngineRequest: LocationEngineRequest = LocationEngineRequest.Builder(1000L)
+            .setPriority(LocationEngineRequest.PRIORITY_HIGH_ACCURACY)
+            .build()
+    ): MapboxNavigation {
+        synchronized(MapboxNavigationProvider::class.java) {
+            mapboxNavigation?.onDestroy()
+            mapboxNavigation = MapboxNavigation(
+                context,
+                navigationOptions,
+                locationEngine,
+                locationEngineRequest
+            )
+
+            return mapboxNavigation!!
+        }
+    }
+
+    /**
+     * Get MapboxNavigation instance.
+     * Should be called after [createNavigation]
+     */
+    @JvmStatic
+    fun navigation(): MapboxNavigation {
```

Comment:
Same as above. What about calling this `retrieve` (or even `get` 😬 although personally not a big fan of getter / setter nomenclature as I believe it should be used only for POJOs / DTOs). Would love to come up with something that reads 👌 but it's not redundant. Naming and designing good APIs is hard... 😅